sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchResponse.java

```diff
@@ -0,0 +1,219 @@
+        return this.responseStatus >= 200 && this.responseStatus <= 299;
+    }
+
+    /**
+     * Gets the activity ID that identifies the server request made to execute the batch.
+     *
+     * @return the activity ID that identifies the server request made to execute the batch.
+     */
+    public String getActivityId() {
+        return this.responseHeaders.get(HttpConstants.HttpHeaders.ACTIVITY_ID);
+    }
+
+    /**
+     * Gets the reason for the failure of the batch request, if any, or {@code null}.
+     *
+     * @return the reason for the failure of the batch request, if any, or {@code null}.
+     */
+    public String getErrorMessage() {
+        return this.errorMessage;
+    }
+
+    /**
+     * Gets the request charge for the batch request.
+     *
+     * @return the request charge measured in request units.
+     */
+    public double getRequestCharge() {
+        final String value = this.responseHeaders.get(HttpConstants.HttpHeaders.REQUEST_CHARGE);
+        if (StringUtils.isEmpty(value)) {
+            return 0;
+        }
+
+        try {
+            return Double.valueOf(value);
+        } catch (NumberFormatException e) {
+            logger.warn("INVALID x-ms-request-charge value {}.", value);
+            return 0;
+        }
+    }
+
+    /**
+     * Gets the response status code of the batch request.
+     *
+     * @return the response status code of the batch request.
+     */
+    public int getResponseStatus() {
+        return this.responseStatus;
+    }
+
+    /**
+     * Gets the response headers.
+     *
+     * @return the response header map.
+     */
+    public Map<String, String> getResponseHeaders() {
+        return this.responseHeaders;
+    }
+
+    /**
+     * Gets the amount of time to wait before retrying this or any other request due to throttling.
+     *
+     * @return the amount of time to wait before retrying this or any other request due to throttling.
+     */
+    public Duration getRetryAfter() {
+        if (this.responseHeaders.containsKey(HttpConstants.HttpHeaders.RETRY_AFTER)) {
+            return Duration.parse(this.responseHeaders.get(HttpConstants.HttpHeaders.RETRY_AFTER));
+        }
+
+        return null;
+    }
+
+    public Integer getSubStatusCode() {
+        return this.subStatusCode;
+    }
+
+    /**
+     * Get all the results of the operations in batch.
+     *
+     * @return Results of operation in batch.
+     */
+    public List<TransactionalBatchOperationResult<?>> getResults() {
+        return this.results;
+    }
+
+    /**
+     * Gets the result of the operation at the provided index in the batch.
+     *
+     * @param index 0-based index of the operation in the batch whose result needs to be returned.
+     *
+     * @return Result of operation at the provided index in the batch.
+     */
+    public TransactionalBatchOperationResult<?> get(int index) {
+        return this.results.get(index);
+    }
+
+    public boolean isEmpty() {
```

Comment:
you have a check if the transactional batch request size is non zero. So we never can get a response with zero length. correct?

if so please remove this method as it is redundant.

Reply:
Agreed. I am checking it too. Cool. Will remove this.